docs/blueprints/forgotten-deletes.md

```diff
@@ -0,0 +1,124 @@
+
+- Why the indirection from piece ID to signed delete requests?
+
+    It is possible for us to store the signed delete request directly at the piece ID path. However, due to the fact that each request may contain multiple piece IDs, we would be replicating the data at each piece ID path. If each individual piece for deletion had its own signature we would be able to directly store the data without a problem.
+
+- Satellite signs a delete request for each piece ID
+
+   Currently the satellite signs a batch of piece IDs for deletion. If instead the satellite signs each individual piece ID, this would make storing and retrieving proof of deletion much easier, as mentioned above. However, the extra overhead of signing each piece may be prohibitive.
+
+- Store pointer to signed proof at piece location in 'blobs' directory
+
+    It may be possible for us to store the pointer to the signed proof at the piece location in the blobs directory rather than a separate 'deleted-pieces' directory. In this case we would need to implement some way of distinguishing deleted pieces from regular pieces, such as a special file extension. The issue I'm seeing is that a fundamental component of the 'pieces' system is the storage.FormatVersion. All pieces have a specific format version which tells us how to interpret the data. Format versions start at 0 and increment as new versions are added. Should deleted pieces then constitute a new format version? It doesn't look like we should do that. When creating a new piece, it is always created using the highest defined version. We don't want to store newly uploaded pieces with the version for deleted pieces! 
+    Simply leaving the value blank doesn't really solve the problem either, as 0 is a specific version already. So we would need to pass along another value in the code, perhaps a boolean indicating that this is deleted piece data, in addition to the format version in order to process deleted piece data correctly, but I have a feeling this approach will result in clutter. Here's a hacky idea: what if we did define a new format version for deleted piece data, but set it to -1? This would circumvent some of the problems with format versions. Though, I'm not sure if this approach holds any substantial benefits over using a separate directory.
+
+## Implementation
+
+1. Add new field, satellite_signature, to DeletePiecesRequest protobuf
+
+2. Add new field, delete_pieces_request, to PieceDownloadResponse protobuf
+
+3. Add code for satellite to sign DeletePiecesRequests
+
+4. Implement file store for piece IDs and signed delete requests
+
+5. Add code to check for and return PieceDeleteRequest if piece cannot be found during download
+
+6. Add code to satellite to delete object if PieceDownloadResponse contains a valid DeletePiecesRequest
```

Comment:
If there is a DeletePiecesRequest for one node, is it safe to assume the entire object should be deleted?

Also add a note to not penalize nodes for audits where they provide a valid DeletePiecesRequest.